src/Ng/CommandHelpers.cs

```diff
@@ -284,24 +285,69 @@ public static StorageFactory CreateStorageFactory(IDictionary<string, string> ar
             else if (storageType.Equals("Azure", StringComparison.InvariantCultureIgnoreCase))
             {
                 string storageAccountName;
-                if (!arguments.TryGetValue("-storageAccountName", out storageAccountName))
+                string storageKeyValue;
+
+                bool useKeyVault = true;
+
+                string storageContainer;
+                if (!arguments.TryGetValue("-storageContainer", out storageContainer))
+                {
+                    TraceRequiredArgument("-storageContainer");
+                    return null;
+                }
+
+                string storageAccountNameFrame;
+                if (!arguments.TryGetValue("-storageAccountName", out storageAccountNameFrame))
                 {
                     TraceRequiredArgument("-storageAccountName");
                     return null;
                 }
 
-                string storageKeyValue;
-                if (!arguments.TryGetValue("-storageKeyValue", out storageKeyValue))
+                string storageKeyValueFrame;
+                if (!arguments.TryGetValue("-storageKeyValue", out storageKeyValueFrame))
                 {
                     TraceRequiredArgument("-storageKeyValue");
                     return null;
                 }
 
-                string storageContainer;
-                if (!arguments.TryGetValue("-storageContainer", out storageContainer))
+                string vaultName;
+                if (!arguments.TryGetValue("-vaultName", out vaultName))
                 {
-                    TraceRequiredArgument("-storageContainer");
-                    return null;
+                    useKeyVault = false;
+                }
+
+                if (useKeyVault)
+                {
+                    string clientId;
+                    if (!arguments.TryGetValue("-clientId", out clientId))
+                    {
+                        TraceRequiredArgument("-clientId");
+                        return null;
+                    }
+
+                    string certificateThumbprint;
+                    if (!arguments.TryGetValue("-certificateThumbprint", out certificateThumbprint))
+                    {
+                        TraceRequiredArgument("-certificateThumbprint");
+                        return null;
+                    }
+
+                    string validateCertificate = "false";
+                    arguments.TryGetValue("-validateCertificate", out validateCertificate);
+
+                    bool shouldValidateCertificate = validateCertificate == null ? false : validateCertificate.Equals("true", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Nit: Add some line breaks, e.g.: 

```
bool result = condition
    ? true
    : false
```

Rule of thumb is that we should be able to read the code on GitHub without having to scroll right :-)